### PR TITLE
Revert Surefire Version Change, Misc Configuration Cleanup

### PR DESCRIPTION
**Pull request by alzimmermsft**
This PR reverts the Maven surefire version rev (3.0.0-m4 -> 3.0.0-m3) due to issues introduced in the new version where some tests weren't running properly (or at all). Additionally, this PR cleans up some configurations in the parent POMs that weren't in the proper location, data library environment variable configurations in the parent POM instead of the data parent, etc.

### CODE REVIEW

File: sdk/parents/azure-data-sdk-parent/pom.xml [hunk 1]
```diff
@@ -441,6 +443,18 @@
...
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-surefire-plugin</artifactId>
+        <version>3.0.0-M3</version> <!-- {x-version-update;org.apache.maven.plugins:maven-surefire-plugin;external_dependency} -->
+        <configuration>
+          <environmentVariables>
+            <test.mode>${testMode}</test.mode>
+            <KEYVAULT_VAULTURI>https://azure-keyvault-3.vault.azure.net</KEYVAULT_VAULTURI>
+            <KEYVAULT_VAULTURI_ALT>https://azure-keyvault-2.vault.azure.net</KEYVAULT_VAULTURI_ALT>
+          </environmentVariables>
```
Comment (srnagar) [inline sdk/parents/azure-data-sdk-parent/pom.xml:455]: The parent pom had these two additional lines. Those should be copied to this pom as well. 

```
<configurationAPIEnvironmentFound>environmentConfiguration</configurationAPIEnvironmentFound>	
<configurationAPIUseRuntimeFirst>environmentConfiguration</configurationAPIUseRuntimeFirst>
```

**Comment by conniey** [comment]
> tests weren't running properly (or at all)
Is this an issue with the plugin, or our tests? Is there a corresponding issue in the surefire repository?

**Comment by alzimmermsft** [comment]
> 
> 
> > tests weren't running properly (or at all)
> 
> Is this an issue with the plugin, or our tests? Is there a corresponding issue in the surefire repository?

Not completely certain on this, but reverting to the older version fixed the tests running which leads me to believe it is an issue with the plugin.

**Comment by sima-zhu** [comment]
If this only applies to Storage, can we check over the groovy eclipse plugin in Storage?

**Comment by conniey** [comment]
> > > tests weren't running properly (or at all)
> > 
> > 
> > Is this an issue with the plugin, or our tests? Is there a corresponding issue in the surefire repository?
> 
> Not completely certain on this, but reverting to the older version fixed the tests running which leads me to believe it is an issue with the plugin.

I think it would be worth creating an issue in SUREFIRE with some repro, so we can at least track this issue... or it may not be fixed in future releases.

https://issues.apache.org/jira/projects/SUREFIRE/issues

File: sdk/parents/azure-data-sdk-parent/pom.xml (hunk 1 above)
Comment (alzimmermsft) [reply sdk/parents/azure-data-sdk-parent/pom.xml:455]: These were being used in tests for Azure Core, will update those tests to no longer require these configurations.

**Comment by alzimmermsft** [comment]
> 
> 
> If this only applies to Storage, can we check over the groovy eclipse plugin in Storage?

For now it'll be easier to revert back to a known working version rather than finding the magic combination of surefire and the eclipse versions that work, especially since the newer version of surefire doesn't offer any clear benefits once the test printing PR is merged in.

**Comment by alzimmermsft** [comment]
> 
> 
> > > > tests weren't running properly (or at all)
> > > 
> > > 
> > > Is this an issue with the plugin, or our tests? Is there a corresponding issue in the surefire repository?
> > 
> > 
> > Not completely certain on this, but reverting to the older version fixed the tests running which leads me to believe it is an issue with the plugin.
> 
> I think it would be worth creating an issue in SUREFIRE with some repro, so we can at least track this issue... or it may not be fixed in future releases.
> 
> https://issues.apache.org/jira/projects/SUREFIRE/issues

I'll file an issue with steps to repro.

**Comment by alzimmermsft** [comment]
Filed this issue against Surefire, https://issues.apache.org/jira/browse/SUREFIRE-1779.

**Pull request merged**